engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java

```diff
@@ -1387,18 +1387,17 @@ private Table whereInInternal(final Table rightTable, final boolean inclusion,
 
     @Override
     public Table flatten() {
+        if (!flat && !isRefreshing() && rowSet.isFlat()) {
+            // We're already flat, and we'll never update; so we can just return a flat copy
+            final QueryTable copyWithFlat = copy();
+            copyWithFlat.setFlat();
+            return copyWithFlat;
+        }
```

Comment:
Move this into the try-with-resources, so the copy picks up the correct update graph (even if it doesn't really matter for static tables).

Reply:
`copy()` uses its own update graph / execution context setup; that should cover us here? I can still move it into the execution context if we feel like it more correct that way...

Reply:
Yes, you're good.